prefect_azure/container_instance.py

```diff
@@ -0,0 +1,704 @@
+        a given time to stderr.
+
+        Args:
+            client: An initialized `ContainerInstanceManagementClient`
+            container_group: The container group that holds the job container.
+            last_log_time: The timestamp of the last output line already streamed.
+
+        Returns:
+            The time of the most recent output line written by this call.
+        """
+        logs = self._get_logs(client, container_group)
+        return self._stream_output(logs, last_log_time)
+
+    def _get_logs(
+        self,
+        client: ContainerInstanceManagementClient,
+        container_group: ContainerGroup,
+        max_lines: int = 100,
+    ) -> str:
+        """
+        Gets the most container logs up to a given maximum.
+
+        Args:
+            client: An initialized `ContainerInstanceManagementClient`
+            container_group: The container group that holds the job container.
+            max_lines: The number of log lines to pull. Defaults to 100.
+
+        Returns:
+            A string containing the requested log entries, one per line.
+        """
+        container = self._get_container(container_group)
+
+        logs = client.containers.list_logs(
+            resource_group_name=self.resource_group_name,
+            container_group_name=container_group.name,
+            container_name=container.name,
+            tail=max_lines,
+            timestamps=True,
+        )
+
+        return logs.content
+
+    def _stream_output(
+        self, log_content: str, last_log_time: datetime.datetime
+    ) -> datetime.datetime:
+        """
+        Writes each entry from a string of log lines to stderr.
+
+        Args:
+            log_content: A string containing Azure container logs.
+            last_log_time: The timestamp of the last output line already streamed.
+
+        Returns:
+            The time of the most recent output line written by this call.
+        """
+        if not log_content:
+            # nothing to stream
+            return last_log_time
+
+        log_lines = log_content.split("\n")
+
+        last_written_time = last_log_time
+
+        for log_line in log_lines:
+            # skip if the line is blank or whitespace
+            if not log_line.strip():
+                continue
+
+            line_parts = log_line.split(" ")
+            # timestamp should always be before first space in line
+            line_timestamp = line_parts[0]
+            line = " ".join(line_parts[1:])
+
+            try:
+                line_time = dateutil.parser.parse(line_timestamp)
+                if line_time > last_written_time:
+                    self._write_output_line(line)
+                    last_written_time = line_time
+            except dateutil.parser.ParserError as e:
+                self.logger.debug(
+                    "Unable to parse timestamp from Azure log line.",
+                    log_line,
```

Comment:
This will throw an error e.g. `TypeError: not all arguments converted during string formatting`

```suggestion
                    "Unable to parse timestamp from Azure log line: %s",
                    log_line,
```